cmd/osctl/cmd/config.go

```diff
@@ -110,8 +112,26 @@ var configAddCmd = &cobra.Command{
 	},
 }
 
+// configGenerateCmd represents the config generate command.
+var configGenerateCmd = &cobra.Command{
+	Use:   "generate",
+	Short: "Generate a set of configuration files",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		input, err := generate.NewInput("test", []string{"1.2.3.4"})
+		if err != nil {
+			helpers.Fatalf("failed to generate PKI and tokens: %v", err)
+		}
+		data, err := generate.Userdata(generate.TypeInit, input)
+		if err != nil {
+			helpers.Fatalf("failed to generate configuration file for type %s: %v", generate.TypeInit.String(), err)
+		}
```

Comment:
Should we add in the validate call here?

Reply:
So when attempting to unmarshal the generated struct I get:

```
oot@1ba3201bae24:/workspaces/talos/cmd/osctl# ./osctl config generate example 1.2.3.4,1.2.3.5,1.2.3.6
panic: reflect: reflect.Value.Set using unaddressable value [recovered]
        panic: reflect: reflect.Value.Set using unaddressable value

goroutine 1 [running]:
gopkg.in/yaml%2ev2.handleErr(0xc000291c60)
        /go/pkg/mod/gopkg.in/yaml.v2@v2.2.2/yaml.go:249 +0x9a
panic(0x1237160, 0xc0000769d0)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
reflect.flag.mustBeAssignable(0x16)
        /usr/local/go/src/reflect/value.go:234 +0x13a
reflect.Value.Set(0x1332220, 0x0, 0x16, 0x1332220, 0xc0002209c0, 0x16)
        /usr/local/go/src/reflect/value.go:1467 +0x2f
gopkg.in/yaml%2ev2.(*decoder).prepare(0xc000308800, 0xc0001c40e0, 0x1332220, 0x0, 0x16, 0xc000291b48, 0x7ac610, 0x12, 0xc0000571a0)
        /go/pkg/mod/gopkg.in/yaml.v2@v2.2.2/decode.go:302 +0x2c9
gopkg.in/yaml%2ev2.(*decoder).unmarshal(0xc000308800, 0xc0001c40e0, 0x1332220, 0x0, 0x16, 0xc0001c40e0)
        /go/pkg/mod/gopkg.in/yaml.v2@v2.2.2/decode.go:324 +0x6e
gopkg.in/yaml%2ev2.(*decoder).document(0xc000308800, 0xc0001c4000, 0x1332220, 0x0, 0x16, 0x8)
        /go/pkg/mod/gopkg.in/yaml.v2@v2.2.2/decode.go:344 +0x7a
gopkg.in/yaml%2ev2.(*decoder).unmarshal(0xc000308800, 0xc0001c4000, 0x1332220, 0x0, 0x16, 0xc000291c20)
        /go/pkg/mod/gopkg.in/yaml.v2@v2.2.2/decode.go:320 +0x1be
gopkg.in/yaml%2ev2.unmarshal(0xc0006b0800, 0x269f, 0x2800, 0x1332220, 0x0, 0x2800, 0x0, 0x0)
        /go/pkg/mod/gopkg.in/yaml.v2@v2.2.2/yaml.go:148 +0x319
gopkg.in/yaml%2ev2.Unmarshal(...)
        /go/pkg/mod/gopkg.in/yaml.v2@v2.2.2/yaml.go:81
github.com/talos-systems/talos/cmd/osctl/cmd.glob..func6(0x2135360, 0xc00000e560, 0x2, 0x2)
        /workspaces/talos/cmd/osctl/cmd/config.go:137 +0x1d0
github.com/spf13/cobra.(*Command).execute(0x2135360, 0xc00000e500, 0x2, 0x2, 0x2135360, 0xc00000e500)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0x2138080, 0x21429d0, 0x1420970, 0xb)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2ec
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
github.com/talos-systems/talos/cmd/osctl/cmd.Execute()
        /workspaces/talos/cmd/osctl/cmd/root.go:54 +0xe0
main.main()
        /workspaces/talos/cmd/osctl/main.go:10 +0x20
```